sips/qbft_sync.md

```diff
@@ -0,0 +1,106 @@
+
+**Inter-instance syncing**  
+A node receiving f+1 messages with a higher height than his local height will trigger sync highest decided in which the node fetches the highest decided (signed message and justification) from its peers.   
+[link to code](https://github.com/bloxapp/ssv-spec/blob/qbft_sync/qbft/controller.go#L98-L100)
+
+```go
+func (c *Controller) ProcessMsg(msg *SignedMessage) (bool, []byte, error) {
+	// code before
+
+	if msg.Message.Height > c.Height {
+            return false, nil, c.processHigherHeightMsg(msg)
+        }
+	
+	// code after
+}
+
+func (c *Controller) processHigherHeightMsg(msg *SignedMessage) error {
+    added, err := c.HigherReceivedMessages.AddIfDoesntExist(msg)
+    if err != nil {
+        return errors.Wrap(err, "could not add higher height msg")
+    }
+    if added && c.f1SyncTrigger() {
+        return c.network.SyncHighestDecided(c.Identifier)
+    }
+    return nil
+}
+
+// f1SyncTrigger returns true if received f+1 higher height messages from unique signers
+func (c *Controller) f1SyncTrigger() bool {
+    uniqueSigners := make(map[types.OperatorID]bool)
+    for _, msg := range c.HigherReceivedMessages.AllMessaged() {
+        for _, signer := range msg.GetSigners() {
+            if _, found := uniqueSigners[signer]; !found {
+                uniqueSigners[signer] = true
+            }
+        }
+    }
+    return c.Share.HasPartialQuorum(len(uniqueSigners))
+}
+
+```
+
+All received decided messages from syncing will trigger, as usual, the upon decided function.  
+
+_Important comment:_ F+1 messages are required to prevent malicous nodes forcing endless syncing from an honest node. In an F+1 (with max F malicious) there is at least 1 honest node sending a message with height > H.
+
+**Intra-instance syncing**  
+A node starting a new instance will sync get highest round change from peer, fetching the highest round change message (with the highest round) from peers.  
+[Link to code](https://github.com/bloxapp/ssv-spec/blob/qbft_sync/qbft/instance.go#L65-L67)    
+
+```go
+func (i *Instance) Start(value []byte, height Height) {
+    i.startOnce.Do(func() {
+        // code before
+		
+        if err := i.config.GetNetwork().SyncHighestRoundChange(i.State.ID, i.State.Height); err != nil {
+            // log
+        }
+    })
+}
+```
+
+This sync only fetches the highest round change because for every round > FirstRound the following proposal requires a quorum of round change messages.  
+Most times when a committee is "stuck" it's because it's waiting for such a quorum. Syncing directly the highest round change will enable the syncing node to quickly form a quorum and/ or speef up F+1 sync.
+
+_**Continuous get highest round change**_  
+Due to the importance of getting the most recent rc message we employ a continuous get highest round change in certain conditions.    
+The above means that a get highest change round call will be triggered every X^(R-4) until the instance decides.
```

Comment:
`X^(R-4)` interval is producing many messages for round 7 (`3^3`)

What about starting to do this from round 8 (`3^4`) instead of 7? or to increase the exponent `X^(R-3)`?

Reply:
Sure I can change to R-3